optimum/intel/generation/modeling.py

```diff
@@ -403,3 +472,553 @@ def _convert_to_standard_cache(
             )
             for layer_past in past_key_value
         )
+
+
+class TSModelForSeq2SeqLM(OptimizedModel, GenerationMixin):
+    auto_model_class = AutoModelForSeq2SeqLM
+    export_feature = "text2text-generation"
+    main_input_name = "input_ids"
+    base_model_prefix = "torch_script_model"
+
+    def __init__(
+        self,
+        decoder,
+        encoder,
+        config: PretrainedConfig = None,
+        decoder_pkv: Optional[torch.jit.RecursiveScriptModule] = None,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        model_save_dir_2: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        model_save_dir_3: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        use_cache: bool = True,
+        **kwargs,
+    ):
+        self.decoder = decoder
+        self.encoder = encoder
+        self.decoder_pkv = decoder_pkv
+        self.config = config
+        self.model_save_dir = model_save_dir
+        self.model_save_dir_2 = model_save_dir_2
+        self.model_save_dir_3 = model_save_dir_3
+        self.preprocessors = kwargs.get("preprocessors", [])
+        self.use_cache = use_cache
+        self._device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        self.decoder.to(self._device)
+        self.encoder.to(self._device)
+        self.normalized_config = NormalizedConfigManager.get_normalized_config_class(config.model_type)(config)
+        self.model_dtype = kwargs.get("model_dtype", None)
+        self.decoder_start_token_id = None
+        if kwargs.get("decoder_start_token_id", None) is not None:
+            self.decoder_start_token_id = kwargs.get("decoder_start_token_id", None)
+        elif config.decoder_start_token_id is not None:
+            self.decoder_start_token_id = config.decoder_start_token_id
+        elif config.bos_token_id is not None:
+            self.decoder_start_token_id = config.bos_token_id
+
+        if self.decoder_start_token_id is None:
+            logger.warning("Please assign the decoder_start_token_id, it may cause unexpected errors.")
+
+        if is_transformers_version("<=", "4.25.1"):

```

Comment:
This can be removed as `[optimum](https://github.com/huggingface/optimum/blob/v1.8.6/setup.py#L18)` now support `transformers>=4.26.0`